tools/pipelines/templates/build-npm-package.yml

```diff
@@ -407,6 +421,33 @@ extends:
                         inputs:
                           sourceFolder: '${{ parameters.buildDirectory }}/nyc/report'
                           targetFolder: $(Build.ArtifactStagingDirectory)/codeCoverageAnalysis
+                      - task: Bash@3
+                        displayName: Report Code Coverage Comparison
+                        condition: and(succeededOrFailed(), eq('${{ parameters.reportCodeCoverageComparison }}', true), eq(variables['ReportDirExists'], 'true'), eq(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.TargetBranch'], 'main'))
+                        continueOnError: false
+                        env:
+                          ADO_API_TOKEN: $(System.AccessToken)
+                          GITHUB_API_TOKEN: $(githubPublicRepoSecret)
+                          TARGET_BRANCH_NAME: '$(targetBranchName)'
```

Comment:
Should all of these be in quotes?

Reply:
Works just fine like this.

Reply:
@alexvy86 might know more, but I think that if the variable has any characters like `:` in it then the YAML parsing will fail. YAML treats an unquoted string as a string but quotes are required if it contains "special characters" but I don't know the full list. This is probably fine, but the API tokens in particular can sometimes have odd characters in them. We can wait to see if it's a real problem.

Reply:
I think in this case it wouldn't matter. If we were using template expressions (`${{ }}`) then maybe, because those get replaced during the "compilation" (syntax checking, etc) phase of the pipeline, where weird characters could indeed break the YAML structure. `$()` variables get replaced at runtime, I think right when the task is about to execute, so the concerns would be more on the side of "will bash interpret this correctly if it has spaces or weird characters?" and most of the time I think we're fine.

Reply:
Oh, I see we added the quotes. I believe ADO strips the quotes before setting the env variable values, so we end up with `token` and not `"token"`, so we should be fine.